### DISCUSSION TITLE
Is it possible to parallelize the test process for dialect compliance testing?

### DISCUSSION THREAD

**Discussion by chalmerlowe** [sqlalchemy/sqlalchemy#10948] [Usage Questions] [answered]
I am running a series of **sqlalchemy 2.0 compliance tests** for a Dialect (specifically BigQuery).

I run my compliance tests using `pytest` and was attempting to use `pytest-xdist` to spread some of the load across multiple CPUs using flags such as `pytest -n4` or `pytest --numprocesses=auto`.

I am getting different results depending on which flags I throw at `pytest-xdist` but ultimately nothing works. It appears, as best we can tell, that things die at this function that has no implementation:

https://github.com/sqlalchemy/sqlalchemy/blob/b836ca0a3a213dc387e08df3ba340e6ec3298985/lib/sqlalchemy/testing/provision.py#L330C1-L339C6

There is no implementation in sqlalchemy 2.0 for this `create_db` function. The docstring indicates that this is intended for use when doing multi-process testing i.e. "Used when a test run will employ multiple processes, e.g., when run via `tox` or `pytest -n4`."

Are folks successfully running tests across multiple processes and/or multiple cores? If so, did you overload the above function in some way OR did you take some other approach. Would love to see some examples of how to successfully do this.

Thanks!

**Comment by zzzeek** [comment, marked as answer]
SQLAlchemy's own test suite uses pytest-xdist.   

the create_db routine you link to should be customized in your own dialect, here is an example:   

https://github.com/sqlalchemy/sqlalchemy/blob/b836ca0a3a213dc387e08df3ba340e6ec3298985/lib/sqlalchemy/dialects/postgresql/provision.py#L27-L59

**Reply by chalmerlowe** [reply]
Let me give that a try. Thanks.

**Reply by chalmerlowe** [reply]
This helped quite a lot, THANK YOU, very much.
We are able to run the test suite.

**Reply by zzzeek** [reply]
great, that was it. great
